src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const arrayResult = [];
+  let result = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      result = Object.assign(result, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete result[key];
+      }
+    }
 
+    if (action.type === 'clear') {
+      for (const clear in result) {
+        delete result[clear];
+      }
+    }
```

Comment:
[CODE STYLE]: Use switch statement if you have limited amount of conditions. This will make your code cleaner and easier to read. Instead of multiple if statements, you should use a switch statement to handle the different types of actions.